deploy-service/common/src/main/java/com/pinterest/deployservice/handler/EnvironHandler.java

```diff
@@ -333,6 +338,13 @@ public String pause(EnvironBean envBean, String operator) throws Exception {
         EnvironBean updateBean = new EnvironBean();
         updateBean.setEnv_state(EnvState.PAUSED);
         updateStage(envBean, updateBean, operator);
+        String scheduleId = envBean.getSchedule_id();
+        ScheduleBean scheduleBean = scheduleDAO.getById(scheduleId);
+        if (scheduleBean.getState() == ScheduleState.COOLING_DOWN) { 
+            // so that when it resumes, it restarts cooling down process    
+            scheduleBean.setState(ScheduleState.RUNNING); 
+            scheduleDAO.update(scheduleBean, scheduleId);
+        }
```

Comment:
In another thought, maybe we don't need to pause the cooling down process if we pause the deploy. Sorry about the confusion.